platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/status/PlatformStatusComponent.java

```diff
@@ -0,0 +1,127 @@
+import com.swirlds.common.system.status.actions.PlatformStatusAction;
+import com.swirlds.common.system.status.actions.TimeElapsedAction;
+import com.swirlds.common.threading.framework.QueueThread;
+import com.swirlds.common.threading.framework.config.QueueThreadConfiguration;
+import com.swirlds.common.threading.framework.config.QueueThreadMetricsConfiguration;
+import com.swirlds.common.threading.manager.ThreadManager;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+
+/**
+ * An asynchronous component which manages platform status.
+ * <p>
+ * This object wraps a {@link PlatformStatusStateMachine}, which contains the actual state machine logic.
+ */
+public class PlatformStatusComponent implements PlatformStatusGetter, Startable, Stoppable {
+    /**
+     * A source of time
+     */
+    private final Time time;
+
+    /**
+     * The thread to handle incoming {@link PlatformStatusAction}s
+     */
+    private final QueueThread<PlatformStatusAction> handleThread;
+
+    /**
+     * The platform status state machine that is being wrapped by this asynchronous component
+     */
+    private final PlatformStatusStateMachine stateMachine;
+
+    /**
+     * Constructor
+     *
+     * @param platformContext    the platform context
+     * @param time               a source of time
+     * @param threadManager      the thread manager
+     * @param notificationEngine the notification engine
+     */
+    public PlatformStatusComponent(
+            @NonNull final PlatformContext platformContext,
+            @NonNull final Time time,
+            @NonNull final ThreadManager threadManager,
+            @NonNull final NotificationEngine notificationEngine) {
+
+        this.time = Objects.requireNonNull(time);
+        Objects.requireNonNull(threadManager);
+        Objects.requireNonNull(notificationEngine);
+
+        final PlatformStatusConfig config =
+                platformContext.getConfiguration().getConfigData(PlatformStatusConfig.class);
+        this.stateMachine = new PlatformStatusStateMachine(time, config, notificationEngine);
+
+        this.handleThread = new QueueThreadConfiguration<PlatformStatusAction>(threadManager)
+                .setComponent("platform-status")
+                .setThreadName("platform-status-state-machine")
+                .setHandler(this::processStatusAction)
+                .setIdleCallback(this::triggerTimeElapsed)
+                .setMetricsConfiguration(
+                        new QueueThreadMetricsConfiguration(platformContext.getMetrics()).enableBusyTimeMetric())
+                .build();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void start() {
+        handleThread.start();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void stop() {
+        handleThread.stop();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public void processStatusAction(@NonNull final PlatformStatusAction action) {
```

Comment:
This method is being called directly by other parts of the system. The queue is being bypassed.

The method that calls into the state machine should be private and only referenced by the code creating the queue thread.

Reply:
Oops, fixed. I made `processStatusAction` private, and added a new method `registerStatusAction`, which is what other parts of the system call